app/assets/javascripts/app/views/notifications_view.js

```diff
@@ -49,47 +41,57 @@ app.views.Notifications = Backbone.View.extend({
     this.updateView(data["guid"], type, data["unread"]);
   },
 
+  /* TODO: Test this */
+  markAllRead: function(){
+    var self = this;
+    this.getAllUnread().each(function(i, el){
+      self.setRead($(el).data("guid"));
+    });
+  },
+
   updateView: function(guid, type, unread) {
     var change = unread ? 1 : -1,
         all_notes = $('ul.nav > li:eq(0) .badge'),
         type_notes = $('ul.nav > li[data-type=' + type + '] .badge'),
         header_badge = $('#notification_badge .badge_count'),
-        note = $('.stream_element[data-guid=' + guid + ']');
+        note = $('.stream_element[data-guid=' + guid + ']'),
+        markAllRead = $('a#mark_all_read_link'),
+        translationKey = 'read';
 
-    if(unread) {
-      note.removeClass("read").addClass("unread");
-      $(".unread-toggle .entypo", note)
-        .tooltip('destroy')
-        .removeAttr("data-original-title")
-        .attr('title',Diaspora.I18n.t('notifications.mark_read'))
-        .tooltip();
-    }
+    if(unread){ note.removeClass("read").addClass("unread"); }
     else {
+      translationKey = 'unread';
       note.removeClass("unread").addClass("read");
-      $(".unread-toggle .entypo", note)
+    }
+
+    $(".unread-toggle .entypo", note)
         .tooltip('destroy')
         .removeAttr("data-original-title")
-        .attr('title',Diaspora.I18n.t('notifications.mark_unread'))
+        .attr('title',Diaspora.I18n.t(translationKey))
```

Comment:
Mixed double and single quotes.<br>'Diaspora' is not defined.<br>Too many errors. (70% scanned).